### PR TITLE
Fix/1001 camelcase warnings

### PR DESCRIPTION
**Pull request by Rykus0**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

contributes to #1001 by addressing React console warnings

## What is the new behavior?
Warnings are no longer displayed when assigning `styles` object to the `style` prop
Converts hyphenated selectors in nested styles to camelCase
Camel Case selectors are then converted back to kebab case for `cssProps` string
Updates references to the changed selectors in JS APIs

## Other information
Will require instructions that the `styles` object won't work directly with styled-components and others (emotion), and that only `cssProps` should be used.

### CODE REVIEW

**Comment by caseybaggz** [comment]
@Rykus0 I don't think it will "require instructions" about how to use the APIs. People who use the 3rd party lib will know what syntax it supports - we don't need to dictate and manage that on our side.

File: packages/headless-styles/sandbox/src/components/Tabs.jsx [hunk 1]
```diff
@@ -138,7 +138,7 @@ function TabJS(props) {
       style={{
         ...tabPropsJS.tab.styles,
         ...(data['aria-selected'] &&
-          tabPropsJS.tab.styles["&[aria-selected='true']"]),
+          tabPropsJS.tab.styles["&[ariaSelected='true']"]),
```
Comment (caseybaggz) [inline packages/headless-styles/sandbox/src/components/Tabs.jsx:141]: I thought that the camelCasing is only valid on pseudo classes, not actual aria & data selectors?

**Comment by Rykus0** [comment]
> I don't think it will "require instructions" about how to use the APIs. People who use the 3rd party lib will know what syntax it supports - we don't need to dictate and manage that on our side.

@caseybaggz I meant mainly that `styles` wouldn't work directly anymore.  Probably a description (or updated description) of the format of the object would be better; make sure we're providing the information for people to make their decision.

**Review by caseybaggz — COMMENTED** [review]
Waiting on sandbox validation.

**Comment by Rykus0** [comment]
Sandbox for Buttons: https://codesandbox.io/s/ps-react-typescript-forked-kz2u4e

**Comment by Rykus0** [comment]
MUI button sandbox: https://codesandbox.io/s/basic-mui-forked-mqour0?file=/src/App.tsx

**Review by caseybaggz — APPROVED** [review]

**Pull request merged**